app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.build(comment_params)
+    @comment.user = current_user
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @post.author_id, post_id: @post.id, id: @post.id)
+      @post.update(comments_counter: @post.comments.count)
+    else
+      redirect_to new_user_post_comment_path(user_id: @post.author_id, post_id: @post.id)
+    end
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```

Comment:
- [ ] [optional] @cosywasswa It would be nice if you made `comment_params` private, this will ensure you have `strong parameters` in your controller that will lead to a more secure application (preventing  against mass assignment vulnerabilities, which can occur when a user sends unexpected parameters to a controller action)